### PR TITLE
Feat/get players pov zappy gui

### PR DESCRIPTION
**Pull request by Thomaltarix**
I implement some huge features:
- The Hitbox detection and printing (only when debug is on)
- The clicks on the players to take their pov
- Implementation of the rotation of the hitboxes
- Add binds to change player pov
- Add bind to go free camera

At the moment, the free cam bind is `Space`
You can left click onto players to get their POV.
When you are in a player POV, you can left and right click to change POV (swaping through teams and players).

To see more details, check this issue: #81

### CODE REVIEW

File: gui/src/Event/Event.cpp [hunk 1]
```diff
@@ -48,3 +62,122 @@ void Gui::Event::switchDisplayDebug()
...
+    for (auto &team : _gameData->getTeams()) {
+        for (auto &player : team.getPlayers()) {
+            if (_render.get()->getIsDebug()) {
+                std::vector<BoundingBox> bboxes = team.getPlayerBoundingBoxes(player.getPosition(), player.getOrientation(), player.getCenterPosition());
+                std::vector<RayCollision> hitbox = team.getPlayerModelHitbox(player.getId(), *_render.get()->getCamera().get());
+
+                for (size_t i = 0; i < bboxes.size(); i++) {
+                    if (hitbox[i].hit) {
+                        DrawBoundingBox(bboxes[i], RED);
+                    }
```
Comment (Njord201) [inline gui/src/Event/Event.cpp:109]: Brackets

File: gui/include/Event/Event.hpp [hunk 2]
```diff
@@ -94,4 +115,52 @@ class Gui::Event {
...
+        */
+        void handleLeftClick();
+
+        /**
+         * @brief Handle the right click.
+        */
+        void handleRightClick();
+
+        /**
+         * @brief Select the player pov;
```
Comment (mathieurobert1) [inline gui/include/Event/Event.hpp:134]:
```suggestion
         * @brief Select the player pov
```

File: gui/src/GameDatas/Team.cpp [hunk 3]
```diff
@@ -108,3 +110,121 @@ void Gui::Team::setEggModelPath(const std::string &eggModelPath)
 {
     _eggModel = LoadModel(eggModelPath.c_str());
 }
+
+std::vector<BoundingBox> Gui::Team::getPlayerBoundingBoxes(std::pair<size_t, size_t> pos, size_t orientation, Vector3 center)
+{
+    std::vector<BoundingBox> bboxes;
+    Model model = this->getPlayerModel();
+
+    for (int i = 0; i < model.meshCount; i++) {;
```
Comment (Njord201) [inline gui/src/GameDatas/Team.cpp:119]: Remove the ;

**Review by Njord201 — CHANGES_REQUESTED** [review]
Some little requests

File: gui/src/Event/Event.cpp [hunk 4]
```diff
@@ -48,3 +62,122 @@ void Gui::Event::switchDisplayDebug()
...
+void Gui::Event::switchPovLeft()
+{
+    _render->setCameraType(Gui::UserCamera::FREE);
+}
+
+void Gui::Event::switchPovRight()
+{
+    _render->setCameraType(Gui::UserCamera::POV_PLAYER);
+    UpdateCamera(_render->getCamera().get(), CAMERA_THIRD_PERSON);
+}
```
Comment (mathieurobert1) [inline gui/src/Event/Event.cpp]: Delete this useless methods

File: gui/include/Event/Event.hpp [hunk 5]
```diff
@@ -74,7 +82,20 @@ class Gui::Event {
         std::unordered_map<KeyboardKey, std::function<void()>> _eventsKeyPressed =
         {
             {KEY_THREE, [this](){switchDisplayDebug();}},
-            {KEY_F3, [this](){switchDisplayDebug();}}
+            {KEY_F3, [this](){switchDisplayDebug();}},
+            {KEY_ONE, [this](){switchPovLeft();}},
+            {KEY_TWO, [this](){switchPovRight();}},
```
Comment (mathieurobert1) [inline gui/include/Event/Event.hpp]: Delete this useless methods

File: gui/src/Event/Event.cpp (hunk 1 above)
Comment (Thomaltarix) [reply gui/src/Event/Event.cpp:109]: Done

File: gui/include/Event/Event.hpp [hunk 6]
```diff
@@ -94,4 +115,52 @@ class Gui::Event {
...
+         * @brief Switch of pov by left.
+         *
+         */
+        void switchPovLeft();
+
+        /**
+         * @brief Switch of pov by right.
+         *
+         */
+        void switchPovRight();
```
Comment (mathieurobert1) [inline gui/include/Event/Event.hpp]: Delete this useless methods

**Review by Njord201 — APPROVED** [review]

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some request

File: gui/include/Event/Event.hpp (hunk 2 above)
Comment (Thomaltarix) [reply gui/include/Event/Event.hpp:134]: Done

File: gui/src/Event/Event.cpp (hunk 4 above)
Comment (Thomaltarix) [reply gui/src/Event/Event.cpp]: Done

File: gui/include/Event/Event.hpp (hunk 5 above)
Comment (Thomaltarix) [reply gui/include/Event/Event.hpp]: Done

File: gui/include/Event/Event.hpp (hunk 6 above)
Comment (Thomaltarix) [reply gui/include/Event/Event.hpp]: Done

**Review by mathieurobert1 — APPROVED** [review]
Nice, a lot of commits my friend !

**Pull request merged**